userspace/libscap/engine/savefile/scap_savefile.c

```diff
@@ -119,6 +119,7 @@ static int32_t scap_read_proclist(scap_reader_t* r, uint32_t block_length, uint3
 		tinfo.exe_ino = 0;
 		tinfo.exe_ino_ctime = 0;
 		tinfo.exe_ino_mtime = 0;
+		tinfo.exe_from_memfd = false;

```

Comment:
Looks like a whole chunk is missing here: we never load the flag even though we store it.

(also, I'd consider stuffing this together with exe_upper_layer in the scap file)

Reply:
Isn't this the part where we load? => [here](https://github.com/falcosecurity/libs/pull/1066/files/b5a9293810660842abd6cbf9264490e99e47e212#diff-7b4e2af897ce4b8645f4f862762f68425d8394805c45c194255730e99078767dR696)
Also, we can't stuff fields in the middle, they should only be appended, else they break backwards compatibility.

Reply:
No, that part literally says `scap_dump_write`.

> Also, we can't stuff fields in the middle, they should only be appended, else they break backwards compatibility.

I know, but the exe_upper_layer uses one bit out of 8, there are 7 remaining where we can put other flags (I'd rename the field while we're at it though). On second thought, it wouldn't be such a great idea since any non-zero bit would be interpreted as exe_upper_layer by older versions. NVM.

Reply:
Fixed